onebusaway-android/src/main/java/org/onebusaway/android/util/RouteColors.kt

```
@@ -17,17 +17,45 @@
+ * canonical parse used by the wire DTO color readers ([org.onebusaway.android.api.colorArgb]).
+ *
+ * OBA hands over a bare hex ("FDB71A") and OTP does too, but not every producer is that disciplined, so
+ * a leading '#' is tolerated here rather than at each call site — this function is the one place that
+ * decides what a route color on the wire may look like, and callers were otherwise obliged to strip the
+ * '#' themselves before handing it over, which several independently did.
  */
 fun parseObaHexColor(hex: String?): Int? = hex?.takeIf { it.isNotEmpty() }?.let {
     try {
-        "#${it.trim()}".toColorInt()
+        "#${it.trim().removePrefix("#")}".toColorInt()
     } catch (e: IllegalArgumentException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)